mp/src/game/shared/momentum/mom_rocket.cpp

```diff
@@ -147,7 +189,11 @@ void CMomRocket::Destroy(bool bNoGrenadeZone)
     SetTouch(NULL);
     AddEffects(EF_NODRAW);
     StopSound("Missile.Ignite");
-    DestroyTrail();
+
+    if (!UseTFTrail())
+    {
+        DestroyTrail();
+    }
```

Comment:
I'm seeing this check before the method a ton.

1. Move the check inside DestroyTrail and just early return there if `UseTfTrail()` is false.
2. Does the tf2 particle get removed automatically when the rocket is removed? Otherwise we may have a leak here.

Reply:
The TF2 particle is just being created via a `ParticleProp()` call, I'm unsure if you'd need to make an extra call to kill the particle when the rocket is killed. I have seen some other entities in TF2 use `ParticleProp()->StopEmission();` in their destructor so I imagine I could just add that?

Reply:
Yes I would do that. Also inside `DestroyTrail` as well (instead of early returning, make it an if/else)

Reply:
`ParticleProp()` is a client function and since `DestroyTrail` is inside an `#ifdef GAME_DLL` block I can't seem to call it in there.

Reply:
Then destructor is fine.